airbyte-webapp/src/hooks/services/ConnectionEdit/ConnectionEditService.tsx

```diff
@@ -46,7 +46,11 @@ const useConnectionEdit = ({ connectionId }: ConnectionEditProps): ConnectionEdi
   const [catalog, setCatalog] = useState<ConnectionCatalog>(() => getConnectionCatalog(connection));
   const [schemaHasBeenRefreshed, setSchemaHasBeenRefreshed] = useState(false);
 
-  const [{ loading: schemaRefreshing, error: schemaError }, refreshSchema] = useAsyncFn(async () => {
+  const {
+    isLoading: schemaRefreshing,
+    error: schemaError,
+    mutateAsync: refreshSchema,
+  } = useMutation<void, SchemaError>(async () => {
```

Comment:
Attempted to replace useAsyncFn to fix the delayed hiding problem but it didn't fix it. Decided to keep this change since useAsyncFn has issues anyway, and we get a typed Error.